### PR TITLE
PICMI analytic applied fields

### PR DESCRIPTION
**Pull request by dpgrote**
Added picmi implementation of several applied fields, including constant field, analytic field, and reflecting mirror. This is dependent on PR28 in the pimcistandard.

### CODE REVIEW

File: Python/pywarpx/picmi.py [hunk 1]
```diff
@@ -555,6 +555,62 @@ def initialize_inputs(self, laser):
...
+
+class Mirror(picmistandard.PICMI_Mirror):
+    def initialize_inputs(self):
+        try:
+            pywarpx.warpx.num_mirrors
+        except AttributeError:
+            pywarpx.warpx.num_mirrors = 0
+            pywarpx.warpx.mirror_z = []
+            pywarpx.warpx.mirror_z_width = []
+            pywarpx.warpx.mirror_z_npoints = []
```
Comment (LDAmorim) [inline Python/pywarpx/picmi.py:606]:
```suggestion
            pywarpx.warpx.num_mirrors = 0
            pywarpx.warpx.mirror_z = []
            pywarpx.warpx.mirror_z_width = []
            pywarpx.warpx.mirror_z_npoints = []
```
This is just a minor question/suggestion.
I think that these are the defaults already on WarpX.
I would say that if the user doesn't write explicitly `pywarpx.warpx.num_mirrors = 0`, then there is no need to do this initialization in PICMI.
Or is it incompatible with PICMI defaults?

**Review by LDAmorim — COMMENTED** [review]
Thanks for this PR @dpgrote !
All looks well to me, but I didn't test it yet.

File: Python/pywarpx/picmi.py (hunk 1 above)
Comment (dpgrote) [reply Python/pywarpx/picmi.py:606]: This is creating the input variables in the picmi implementation that will be filled by the user's value below. Without this code, those Python variables would not be created anywhere. If a Mirror object is never used, those variables, num_mirrors etc, would not be created and would not appear in the input for WarpX. Does that help?

Comment (LDAmorim) [reply Python/pywarpx/picmi.py:606]: Yes, thank you @dpgrote !

**Review by RemiLehe — APPROVED** [review]

**Pull request merged**